unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -266,12 +283,21 @@ public void actionFinished(bool success, System.Object actionReturn=null)
             }
 
             lastActionSuccess = success;
-			this.actionComplete = true;
+			this.agentState = newState;
 			this.actionReturn = actionReturn;
 			actionCounter = 0;
 			targetTeleport = Vector3.zero;
+
+        }
+
+		public virtual void actionFinished(bool success, System.Object actionReturn=null) 
```

Comment:
Why are there 2 `actionFinished` here? surprised the type system is not confused as I am of which one is being called from `actionFinishedEmit`

Reply:
I see now, backcompat